### DISCUSSION TITLE
Open automatically *.code-workspace files automatically in VSCode Desktop

### DISCUSSION THREAD

**Discussion by beagleknight** [coder/coder#9435] [General] [answered]
Hi there,

we are trying to automatically open a *.code-workspace file when we open VSCode Desktop using the corresponding button.

We have seen in the source code that this button is rendered using the `VSCodeDesktopButton` component and that component is opening a url that includes the `agent.expanded_directory`. 

Is there a way to configure this so the `expanded_directory` points to our *.code-workspace file?

Best regards,

**Comment by matifali** [comment]
you can set [`dir`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/agent#dir) in `coder_agent` to open VS code in that directory.

**Reply by beagleknight** [reply]
Hey @matifali !

I just tried your suggestion, and it doesn't work 😞 . I see the following errors when I try to open the VSCode Desktop:

```
local-server-1> ssh child died, shutting down

[10:26:56.682] Failed to parse remote port from server output
```

Reading the documentation, we see that the `dir` field setups the starting directory of the session, so we cannot use the `*.code-workspace` path here.

**Reply by matifali** [reply]
Yes this only allows you to open a fixed specific directory. That's how Vs code works. You can open a specific folder while starting it.

**Reply by matifali** [reply]
For example see this example for Coder dogfood template. This allows us to open VS Code in our repo directory. 
https://github.com/coder/coder/blob/ed50acaabf5dc7a5048f6c5e3420d963d2502d21/dogfood/main.tf#L89

**Reply by beagleknight** [reply]
I understand! 👍 The issue here is we want to open a VS Code workspace from that directory (such as `workspace-coder.code-workspace`). Right now when VS Code opens it asks us to open the workspace clicking a button, but it would be nice if we can do this automatically.

We have been playing with the URL used by the `VSCodeDesktopButton` component (the one that starts with `vscode://`) and tweaking the `folder` query parameter to include the whole path to our workspace file do the trick. Do you think this could be configured somehow?

**Reply by matifali** [reply]
You can check here if the vscode URI supports opening a specific workspace. https://github.com/microsoft/vscode-uri. I hope it helps

**Reply by matifali** [reply, marked as answer]
Yes it should work @beagleknight. For better control you can switch to vscode-desktop module. 
https://registry.coder.com/modules/vscode-desktop

**Comment by matifali** [comment]
@code-asher do we support opening workspace files from our vscode plugin URI?

**Reply by code-asher** [reply]
We are not handing workspace files explicitly but I tried setting `folder` to a `.code-workspace` file and it seems to work, actually.
